src/reports.cpp

```diff
@@ -370,10 +370,11 @@ static config unit_alignment(reports::context & rc, const unit* u, const map_loc
 {
 	if (!u) return config();
 	std::ostringstream str, tooltip;
-	const std::string align = unit_type::alignment_description(u->alignment(), u->gender());
-	const std::string align_id = unit_alignments::get_string(u->alignment());
+	bool alignment_abilitie = ((under_alignment_abilities(*u, hex, 0).second).empty()) ? false : true;
+	const std::string align = alignment_abilitie ? translation::dsgettext("wesnoth", (under_alignment_abilities(*u, hex, 0, false, true).second).c_str()) : unit_type::alignment_description(u->alignment(), u->gender());
+	const std::string align_id = alignment_abilitie ? under_alignment_abilities(*u, hex, 0).second : unit_alignments::get_string(u->alignment());
```

Comment:
Using the ternary operator here makes the lines long and hard to read. As a general rule, I'd prefer removing the `const` from the existing lines, and then adding a conditional block for the rare case that an ability is in effect:

```
auto align = unit_type::alignment_description(u->alignment(), u->gender());
auto align_id = unit_type::get_string(u->alignment());
if(alignment_abilitie) {
    align = ... ;
    align_id = ...;
}
```

However, in this case I think `under_alignment_abilities` should return the alignment using the `unit_alignments` enum, and the existing lines then use that instead of `u->alignment()`:
```
const auto effective_alignment = under_alignment_abilities(...);
const auto align = unit_type::alignment_description(effective_alignment, u->gender());
const auto align_id = unit_type::get_string(effective_alignment);
```

Returning the enumerated type would also mean that `under_alignment_abilities` wouldn't need to handle the gender at all - this is the only caller that uses it, and it can do it here via `alignment_description`.

Reply:
i think removeunit_alignment abilities for now and wait another PR.